app/services/budgetkey-item.ts

```diff
@@ -46,22 +46,42 @@ export class BudgetKeyItemService {
     Promise.reject(new Error('No layout for ' + path));
   }
 
-  getItemData(query: string, headersOrder: string[]): Promise<object> {
-    let url = 'http://next.obudget.org/api/query?query=' +
-      encodeURIComponent(query);
+  private _budgetNumberFormatter(value: any) {
+    return parseFloat(value).toLocaleString('he-IL', { style: 'currency', currency: 'ILS' });
+  }
+
+  private _budgetLinkFormatter(value: string, hLink: string) {
+    return '<a>' + value + '</a>';
+  }
+
+  getItemData(query: string, headersOrder: string[], formatters: string[]): Promise<object> {
+    const url = 'http://next.obudget.org/api/query?query=' + encodeURIComponent(query),
+    formatterKeys = _.keys(formatters),
+    formatterValues = _.values(formatters);
```

Comment:
who guarantees that order of formatterKeys and formatterValues is the same (i.e. key matches value)?

Reply:
I don't know. What do you suggest?

Reply:
I think that after code simplification this won't be necessary anyway.

Reply:
So?...

Reply:
See these comments:
https://github.com/OpenBudget/budgetkey-app-generic-item/pull/41/files/596f780c804dc1ba206f5c064d87276e2f907ac3#diff-f8ee414efc5e5952dbdb531f63699137
and
https://github.com/OpenBudget/budgetkey-app-generic-item/pull/41#discussion_r159365962

Reply:
I think I got it. take a look after next fix